Fixed arraySort.js

r4h33q opened the pull request:
## Type
- [X] Bug Fix
- [X] Functions: $arraySort
- [ ] Callbacks: \<Callback Name>
- [ ] Handlers: \<Handler Type>
- [ ] Others: \______

Dependencies (Third Party Modules) needed: <Name | Github Link> (Maximum: 20MB~ size)

Want a credit? Discord tag or other social media link: lightner.

Referenced Issue: Only reported in Discord (Answer for an issue, if any)

## Description
Fix $arraySort to actually sort array instead of treating each value as equal

r4h33q committed:
Fixed arraySort.js

Faf4a commented on src/functions/arraySort.js:
```diff
@@ -9,7 +9,7 @@ module.exports = async (d) => {
   }
 
   d.arrays[name] = d.arrays[name].sort((a, b) =>
-    type === "asc" ? a - b : b - a,
+    type === "asc" ? (a < b ? -1 : a > b ? 1 : 0) : (a < b ? 1 : a > b ? -1 : 0),
   );
```
Maybe doing it like this:

```js
.sort((a, b) =>
    typeof a === "number" && typeof b === "number" ? a - b :
    typeof a === "string" && typeof b === "string" ? a.localeCompare(b) :
    typeof a === "number" ? -1 : 1
);
```

is more readable and robust? you could use .reverse() at the end to reverse the array if type is desc

Faf4a reviewed (changes requested):
thanks for the pr! please take a look at the review I left behind

r4h33q committed:
Fix arraySort.js
Attempt 2 at fixing this
This way of sorting the array emulates the same way that arrays are sorted in javascript

r4h33q committed:
Oops
Forgot to delete that part

r4h33q commented:
I didn't exactly follow the way you said to do it in the comment because this way of sorting the array emulates the same way that arrays are sorted in javascript

Leref committed:
Merge branch 'v6' into fix-arraySort
